azerothcore/azerothcore-wotlk
# fix(DB/Smart_scripts): Add event on quest completion for last part of Elixir of Agony

## Changes Proposed:
- Added small event on quest completion on the last part of Elixir of Agony chain.

## Issues Addressed:
- Tarren Mill's Elixir of Agony quest keg doesn't disappear.
- Captured farmers arent scripted they just stand there.
(Didn't find an open issue, I discovered this on ChromieCraft)



## SOURCE:

https://www.youtube.com/watch?v=qRzLyUtezwU
https://www.youtube.com/watch?v=Png8rIiAxcM&t=0s

## Tests Performed:

- Tested without the changes on quest completion Elixir of Agony(524) Captured Farmers are static and Tainted keg doesn't disappear.
![2021-04-19 13-56-22](https://user-images.githubusercontent.com/10816895/115241215-3b90ab80-a129-11eb-821e-39807d887dba.gif)
- Tested in game on linux/docker setup from windows client.
- Tested with the changes. It now plays a small event on quest completion.
![2021-04-19 15-08-14](https://user-images.githubusercontent.com/10816895/115241294-5105d580-a129-11eb-85eb-18e2ded5a67f.gif)

## How to Test the Changes:

Test without changes:
1. Go to Tarren Mill `.go xyz -5.8 -939.1 61.9 [267]`
2. Accept the last part of Elixir of Agony quest `.quest add 524`
3. Complete the quest by using the dusty rug on the floor.
4. Check if nothing happens after you turn in the quest and tainted keg is already present.

Test with changes:
1. Apply the query from `rev_1618834215629926428.sql`.
2. Reload/restart your server.
3. Go to Tarren Mill `.go xyz -5.8 -939.1 61.9 [267]`
4. Remove the quest if you have completed it already `.quest remove 524`.
4. Accept the last part of Elixir of Agony quest `.quest add 524`.
5. Check if the npc's act in the same manner as the source videos.

## Known Issues and TODO List:



## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
